libs/platform/src/lib/components/form/select/select/select.component.ts

```diff
@@ -0,0 +1,321 @@
+    providers: [{ provide: FormFieldControl, useExisting: SelectComponent, multi: true }]
+})
+export class SelectComponent extends BaseSelect implements OnInit, AfterViewInit, AfterViewChecked {
+
+    /** Whether the select component is disabled. */
+    @Input()
+    state: SelectControlState = null;
+
+    /** Whether the select component should be displayed in mobile mode. */
+    @Input()
+    mobile = false;
+
+    /** Whether the select component is disabled. */
+    @Input()
+    stateMessage: string;
+
+    /** Whether the select component is disabled. */
+    @Input()
+    disabled = false;
+
+    /** Whether the select component is readonly. */
+    @Input()
+    readonly = false;
+
+    /** Placeholder for the select. Appears in the triggerbox if no option is selected. */
+    @Input()
+    placeholder: string;
+
+    /** Whether the select is in compact mode. */
+    @Input()
+    compact = false;
+
+    /** @deprecated
+     * it is handled internally by controlTemplate != null|undefined is
+     * Equal as extendedBodyTemplate as true.
+     * Whether option components contain more than basic text. */
+    @Input()
+    extendedBodyTemplate = false;
+
+    /** Max height of the popover. Any overflowing elements will
+     *  be accessible through scrolling. */
+    @Input()
+    maxHeight: string;
+
+    /** Glyph to add icon in the select component. */
+    @Input()
+    glyph = 'slim-arrow-down';
+
+    /** Whether close the popover on outside click. */
+    @Input()
+    closeOnOutsideClick = true;
+
+    /**
+     * Preset options for the Select body width, whatever is chosen, the body has a 600px limit.
+     * * `at-least` will apply a minimum width to the body equivalent to the width of the control. - Default
+     * * `equal` will apply a width to the body equivalent to the width of the control.
+     * * 'fit-content' will apply width needed to properly display items inside, independent of control.
+     */
+    @Input()
+    fillControlMode: PopoverFillMode = 'at-least';
+
+    /** Custom template used to build control body. */
+    @Input()
+    controlTemplate: TemplateRef<any>;
+
+    /** The element to which the popover should be appended. */
+    @Input()
+    appendTo: ElementRef;
+
+    /**
+     * If the option should be unselected and value changed to undefined, when the current value is
+     * not presented in option array. Switching it off can be handy, when there is some delay between providing
+     * possible options and chosen value.
+     */
+    @Input()
+    unselectMissingOption = true;
+
+    /** Time to wait in milliseconds after the last keydown
+     *  before focusing or selecting option based on alphanumeric keys. */
+    @Input()
+    typeaheadDebounceInterval = 250;
+
+    /** Binds to control aria-labelledBy attribute */
+    @Input()
+    ariaLabelledBy: string = null;
+
+    /** Sets control aria-label attribute value */
+    @Input()
+    ariaLabel: string = null;
```

Comment:
Not sure why you have several properties defined both in this class and parent class